src/collects/seashell/backend/asan-error-parse.rkt

```diff
@@ -0,0 +1,113 @@
+#lang typed/racket
+
+;; Use the function "asan-error-parse" to reads the asan error message
+;; and procude a parsed structure
+(provide format-asan-error)
+
+(struct StackFrame ([frame : Integer]
+                    [file : String]
+                    [ln : Integer]
+                    [func : String])
+  #:transparent)
+(define-type ErrorComponent (U String StackFrame))
+(define-type ASANError (Listof ErrorComponent))
+
+(: format-asan-error (String -> String))
+(define (format-asan-error x)
+  (ASANError->string (asan-error-parse x)))
+
+;; format ASANError in a friendly manner
+(: ASANError->string (ASANError -> String))
+(define (ASANError->string x)
+  (string-join 
+    (map (lambda ([x : ErrorComponent])
+           (match x
+             [(? string?) x]
+             [(StackFrame frame file ln func)
+              (format "frame ~a, file \"~a\", line ~a, function \"~a\"" frame file ln func)]))
+         x) 
+    "\n"
+    #:after-last "\n"))
+
+;; The main parser function. Use the error message as the argument.
+(: asan-error-parse (String -> ASANError))
+(define (asan-error-parse contents)
+  (define lines : (Listof String) 
+    (filter-map (lambda ([x : String])
+                  (match (string-trim x)
+                    ["" #f]
+                    [x x]))
+                (string-split contents "\n")))
+  (when (< (length lines) 2)
+    (error 'ParseError "Input message is too short.\n Here is an example of an expected input:\n 
+           =================================================================
+           ==33482==ERROR: LeakSanitizer: detected memory leaks
+           Direct leak of 144 byte(s) in 9 object(s) allocated from:
```

Comment:
Can you also parse the extra information (i.e the stack frame layout, location of fault) that ASAN produces?